### PR TITLE
Update TOC

### PR DESCRIPTION
**Pull request by smitpatel**
Putting up for first draft on TOC itself.
Will remove things marked with no page once approved. Add stub pages for them in appropriate folder so we remember in future (but it is ok to deviate)
Updating uid/names to match the structure. Then will add redirections.

### CODE REVIEW

**Comment by smitpatel** [comment]
Internal review URL https://review.docs.microsoft.com/en-us/ef/?branch=pr-en-us-2843

**Review by JeremyLikness — CHANGES_REQUESTED** [review]
See image below - the children for "Entity Framework Core" aren't nested but have moved to the top level
toc

**Review by ajcvickers — APPROVED** [review]

File: entity-framework/toc.yml [hunk 1]
```diff
@@ -18,10 +18,29 @@
...
+    - name: Welcome!
       href: core/index.md
     - name: Take the EF Core 6 survey >>
       href: https://www.surveymonkey.com/r/XRL3JBL
+    - name: Getting started
+      items:
+      - name: EF Core Overview
+        items:
+        #- name: What is EF Core? <<<--- No Page
+        - name: Getting EF Core <<<--- formerly Install EF Core
```
Comment (smitpatel) [inline entity-framework/toc.yml:30]: Can folks verify items with '<<<---" in this file which is not commented out with # at the start?
Some requires verification only. Some did not have a very specific instruction on what to do so requires guidance.

Comment (smitpatel) [reply entity-framework/toc.yml:30]: Should have tagged @dotnet/efteam

Comment (ajcvickers) [reply entity-framework/toc.yml:30]: Install

File: entity-framework/toc.yml [hunk 2]
```diff
@@ -193,7 +173,7 @@
         href: core/querying/client-eval.md
       - name: Tracking vs. no-tracking
         href: core/querying/tracking.md
-      - name: Single vs. split queries
+      - name: Single vs. split queries <<<--- New added after meeting
```
Comment (ajcvickers) [inline entity-framework/toc.yml:176]: Split queries

File: entity-framework/toc.yml [hunk 3]
```diff
@@ -223,8 +204,66 @@
...
+    #  - name: Find and local queries <<<--- No Page
+    #  - name: Accessing entity state <<<--- No Page
+    #  - name: Notification entities <<<--- No Page
+    #  - name: Change tracking proxies <<<--- No Page
+
+    - name: Logging, events, and diagnostics
+      items:
+      - name: Overview <<<--- Moved from events/index.md
+        href: core/logging-events-diagnostics/index.md
+      - name: Logging <<<--- former page with no notes on where to move
```
Comment (ajcvickers) [inline entity-framework/toc.yml:222]: Redirect to logging overview

File: entity-framework/toc.yml [hunk 4]
```diff
@@ -223,8 +204,66 @@
...
+      - name: Asynchronous programming
+        href: core/miscellaneous/async.md
+      - name: Nullable reference types
+        href: core/miscellaneous/nullable-reference-types.md
+      #- name: Using dependency injection <<<--- No Page
+      - name: Collations and case sensitivity
+        href: core/miscellaneous/collations-and-case-sensitivity.md
+      - name: Connection resiliency
+        href: core/miscellaneous/connection-resiliency.md
+      - name: Configuration and initialization <<<--- formerly Configure a DbContext
```
Comment (ajcvickers) [inline entity-framework/toc.yml:260]: Move to top level, "DbContext configuration and initialization"

See [_DbContext Configuration and Initialization_]() for more information.

**Comment by smitpatel** [comment]
This is updated with changes from today's meeting. Re-review?

**Comment by ajcvickers** [comment]
LGTM :shipit:

**Pull request merged**